src/libsystemd-network/sd-dhcp6-lease.c

```diff
@@ -148,6 +154,15 @@ int dhcp6_lease_get_iaid(sd_dhcp6_lease *lease, be32_t *iaid) {
         return 0;
 }
 
+int dhcp6_lease_set_duid(sd_dhcp6_lease *lease, struct duid *duid, size_t duid_len) {
+        assert_return(lease, -EINVAL);
+        assert_return(duid, -EINVAL);
+
+        lease->duid = duid;
+        lease->duid_len = duid_len;
```

Comment:
this looks wrong. You are keeping references to external objects here. That sounds surprising at least, but is most likely wrong. Is this really what you want to do, and shouldn't you rather copy the duid object here?